packages/nimble-components/src/text-field/styles.ts

```diff
@@ -168,6 +170,7 @@ export const styles = css`
     [part='end']::after {
         content: '';
         position: absolute;
+        margin-left: calc(-1 * (${borderWidth} + ${smallPadding}));
```

Comment:
Poking in devtools, the `::after` is an absolutely positioned element. Instead of the negative margin (which I'll admit don't fully understand where this calc is coming from) can we say this this is is just positioned `left: 0px`? Seems like the intention is to have the hover line flush to the left.

Reply:
I agree that `left: 0px` would be more ideal styling, but it breaks the hover animation. The animation is intended to grow from the center of the control, but setting `left: 0px` makes it grow from the left edge.

The negative margin here is to offset the padding of the `.root` element, which had made this element (the hover border) incorrectly shifted to the right.

Reply:
makes sense!